### ISSUE TITLE
distributed joins fail when join alias and column aliases are used

### ISSUE THREAD

**Issue by mtuncer** [citusdata/citus#2012]
Following query
```sql
select * from (select k1, random()
from (users_table join events_table using (user_id)) k (k1, k2, k3)) l;
```
fails with
```sql
column "k1" specified in USING clause does not exist in left table
```
offending worker query is
```sql
SELECT
    worker_column_1 AS k1,
    worker_column_2 AS random
FROM (
    SELECT
        l.k1 AS worker_column_1,
        l.random AS worker_column_2
    FROM (
        SELECT
            k.k1, random() AS random
        FROM (
            public.users_table_102011 users_table
         JOIN
            public.events_table_102015 events_table
         USING (k1)
         ) k(k1, k2, k3, value_2, value_3, val_4, "time", event_type, value_2_1, value_3_1, value_4)
    ) l
) worker_subquery
WHERE true
```

Notes:
1 - join is given an alias
2 - joined columns are also given aliases
3 - join uses ```using``` clause not ``on``
4 - it occurs on both router and multi-shard queries

Preliminary investigation resulted that this is more like a deparsing problem on Postgres side.

**Comment by mtuncer** [comment]
hint : create a view with this query, and try to display view query to ask postgres to deparsa

**Comment by mtuncer** [comment]
I created a view using the same query
```sql
create view vm as
select * from (select k1, random()
from (users_table join events_table using (user_id)) k (k1, k2, k3)) l;
```

```\d+ vm``` shows
```sql
                                  View "public.vm"
 Column |       Type       | Collation | Nullable | Default | Storage | Description 
--------+------------------+-----------+----------+---------+---------+-------------
 k1     | integer          |           |          |         | plain   | 
 random | double precision |           |          |         | plain   | 
View definition:
 SELECT l.k1,
    l.random
   FROM ( SELECT k.k1,
            random() AS random
           FROM (users_table users_table(k1, "time", value_1, value_2, value_3, val_4)
             JOIN events_table events_table(k1, "time", event_type, value_2, value_3, value_4) USING (k1)) k(k1, k2, k3, value_2, value_3, val_4, "time", event_type, value_2_1, value_3_1, value_4)) l;
``` 

So it pushes down column aliases to inner table level and deparsing works fine, however when we run this query in citus, we still get the same error because it looks like when we create shardid appended table id, we do not put updated column name aliases and ```using(k1)``` is left without ```k1``` is not defined.

**Comment by onderkalaci** [comment]
Just realized that we should fix this issue given that might break some users after #2481.

**Comment by metdos** [comment]
> Just realized that we should fix this issue given that might break some users after #2481.

Is this related to #2588?

**Comment by onderkalaci** [comment]
> Is this related to #2588?

Not really, but I just realized this issue should be fixed since #2481 makes outer joins go through query pushdown, so could regress some users.

**Comment by onderkalaci** [comment]
> so, does this mean we can return wrong results?

No, but some outers joins that used to work could give errors like: `column "k1" specified in USING clause does not exist in left table`
